File: backend/src/main/java/org/rent/room/be/service/QuestionService.java
package org.rent.room.be.service;

import org.rent.room.be.dto.request.question.CreateQuestionRequest;
import org.rent.room.be.dto.request.question.UpdateQuestionRequest;
import org.rent.room.be.dto.response.question.QuestionResponse;

import java.util.List;
import java.util.UUID;

public interface QuestionService {
    QuestionResponse createQuestion(CreateQuestionRequest request);

    QuestionResponse getQuestion(UUID questionId);

    List<QuestionResponse> getQuestions(UUID quizId);

    QuestionResponse updateQuestion(UUID questionId, UpdateQuestionRequest request);

    void deleteQuestion(UUID questionId);
}

File: frontend/src/services/questions/questionService.ts
import api from "../../config/axios";
import type { QuestionOptionDto } from "../question-options/questionOptionService";

export type QuestionType = "SINGLE_CHOICE" | "MULTIPLE_CHOICE";

export interface QuestionDto {
  questionId: string;
  quizId: string;
  questionText: string;
  questionType: QuestionType;
  orderIndex: number;
  options?: QuestionOptionDto[];
  createdAt?: string;
  updatedAt?: string;
}

export interface CreateQuestionRequest {
  quizId: string;
  questionText: string;
  questionType: QuestionType;
  orderIndex: number;
}

export interface UpdateQuestionRequest {
  questionText: string;
  questionType: QuestionType;
  orderIndex: number;
}

export const questionService = {
  async getQuestions(quizId?: string): Promise<QuestionDto[]> {
    const res = await api.get("/questions", {
      params: quizId ? { quizId } : undefined,
    });
    return (res.data.result || []) as QuestionDto[];
  },

  async createQuestion(payload: CreateQuestionRequest): Promise<QuestionDto> {
    const res = await api.post("/questions", payload);
    return res.data.result as QuestionDto;
  },

  async updateQuestion(
    questionId: string,
    payload: UpdateQuestionRequest,
  ): Promise<QuestionDto> {
    const res = await api.put(`/questions/${questionId}`, payload);
    return res.data.result as QuestionDto;
  },

  async deleteQuestion(questionId: string): Promise<void> {
    await api.delete(`/questions/${questionId}`);
  },
};
